src/main/java/Sally.java

```diff
@@ -0,0 +1,291 @@
+                    }
+                    printBorder();
+                } else {
+                    throw new SallyException.SallyTaskNotFoundException();
+                }
+            } else {
+
+                //ToDos
+                if (message.startsWith("todo")) {
+                    if (message.length() > 4) {
+                        String description = message.substring(5);
+                        Task.makeTask(description, "", Task.Type.TODO, true);
+                        System.out.println("setelah make task");
+                    } else {
+                        throw new SallyException.SallyNoDescriptionException();
+                    }
+                }
+
+                //Deadlines
+                else if (message.startsWith("deadline")) {
+                    String description, by;
+                    if (message.length() <= 8) {
+                        throw new SallyException.SallyInvalidInputException();
+                    } else if (message.contains("/by ")) {
+                        description = message.substring(9, message.indexOf("/by") - 1);
+                        by = message.substring(message.indexOf("/by") + 4);
+                        LocalDate localDate;
+                        try {
+                            localDate = LocalDate.parse(by);
+                            by = localDate.format(DateTimeFormatter.ofPattern("MMM dd yyy"));
+                        } catch (DateTimeParseException e) {
+                            System.out.println("Oops! Date has to be in the format of yyyy-mm-dd");
+                        }
+                        Task.makeTask(description, by, Task.Type.DEADLINE, true);
+                    } else {
+                        throw new SallyException.SallyNoDeadlineException();
+                    }
+                }
+
+                //Events
+                else if (message.startsWith("event")) {
+                    String description, at;
+                    if (message.length() <= 5) {
+                        throw new SallyException.SallyInvalidInputException();
+                    } else if (message.contains("/at ")) {
+                        description = message.substring(6, message.indexOf("/at") - 1);
+                        at = message.substring(message.indexOf("/at") + 4);
+                        Task.makeTask(description, at, Task.Type.EVENT, true);
+                    } else {
+                        throw new SallyException.SallyNoPlaceException();
+                    }
+                }
+
+                //Any other messages
+                else {
+                    throw new SallyException.SallyInvalidInputException();
+                }
+            }
+        } catch (SallyException e) {
+            System.out.println(e);
+        } finally {
+            messaging();
+        }
+    }
+
+    protected static String printList() {
```

Comment:
Since this function does not actually do any printing, maybe you can rename it to `convertListToString` or something similar?